EasyPost/PaymentMethod.cs

```diff
@@ -1,38 +1,119 @@
 using System;
-using System.Threading.Tasks;
 using Newtonsoft.Json;
-using RestSharp;
 
 namespace EasyPost
 {
+    /// <summary>
+    ///     Represents a summary of the primary and secondary payment methods on the user's account.
+    /// </summary>
     public class PaymentMethod : Resource
     {
+        /// <summary>
+        ///     Payment method priority
+        /// </summary>
+        public enum Priority
+        {
+            Primary,
+            Secondary
+        }
+
         [JsonProperty("id")]
         public string id { get; set; }
         [JsonProperty("object")]
         public string Object { get; set; }
         [JsonProperty("primary_payment_method")]
-        public CreditCard primary_payment_method { get; set; }
+        public PaymentMethodObject primary_payment_method { get; set; }
         [JsonProperty("secondary_payment_method")]
-        public CreditCard secondary_payment_method { get; set; }
+        public PaymentMethodObject secondary_payment_method { get; set; }
+    }
+
+    /// <summary>
+    ///     Represents a credit card or a bank account.
+    ///     Have to collect all possible properties since we can't anticipate which type of payment method the API will return.
+    /// </summary>
+    public class PaymentMethodObject : Resource
```

Comment:
Thought: I was looking over Go this morning and we actually do the same thing there on a couple objects so there is precedent to do this.

Reply:
Type-strictness FTW